app/main/posts/posts-routes.js

```diff
@@ -86,12 +86,24 @@ function (
             }
         }
     )
+
+    /**
+     * @uirouter-refactor
+     * we need to  be able to set mode-context as collectionModeContext or as savedSearchModeContext
+     * without the route since the user would land in /views/map after the redirect
+     */
     .state(
         {
             url: '^/savedsearches/:savedSearchId/data',
             name: 'posts.data.savedsearch',
             onEnter: function ($state, $transition$, PostFilters, savedSearch) {
+                PostFilters.setMode('savedsearch', savedSearch);
```

Comment:
Users need to be able to go to their bookmarked saved searches and see them but we won't BE in the url anymore since it's not supported in general (so we shouldn't have a special "old" URL , but a redirect would be desirable, IMO). The problem is that we don't have a good way that I know of to set the mode context named view based on params. :/ not sure how to go about it.

Reply:
Could you potentially use a component provider to switch mode context component based on some param?`https://ui-router.github.io/ng1/docs/latest/interfaces/ng1.ng1viewdeclaration.html#componentprovider`
Or course that won't update when the filters change, only on state change

ie:
```
             views: {
                  'mode-context': {
                        'componentProvider' : () {
                                 return 'someModeContext';
                         }
                  }
              }
```